assets/js/theme.js

```diff
@@ -0,0 +1,78 @@
+'use strict';
+
+(() => {
+    const LS_THEME_KEY = 'theme';
+    const THEMES = Object.freeze({
+        LIGHT: 'light',
+        DARK: 'dark',
+        AUTO: 'auto'
+    });
+
+    const body = document.body;
+    const config = body.getAttribute("data-theme");
+
+    const getThemeState = () => {
+        const lsState = localStorage.getItem(LS_THEME_KEY);
+        if (lsState) return lsState;
+
+        let state;
+        switch (config) {
+            case THEMES.DARK:
+                state = THEMES.DARK;
+                break;
+            case THEMES.LIGHT:
+                state = THEMES.LIGHT;
+                break;
+            case THEMES.AUTO:
+            default:
+                state = window.matchMedia("(prefers-color-scheme: dark)").matches
+                    ? THEMES.DARK
+                    : THEMES.LIGHT;
+                break;
+        }
+        return state;
+    };
+
+    const initTheme = (state) => {
+        if (state === THEMES.DARK) {
+            document.documentElement.classList.add(THEMES.DARK);
+            document.documentElement.classList.remove(THEMES.LIGHT);
+        } else if (state === THEMES.LIGHT) {
+            document.documentElement.classList.remove(THEMES.DARK);
+            document.documentElement.classList.add(THEMES.LIGHT);
+        }
+    };
+
+    // init theme ASAP, then do the rest.
+    initTheme(getThemeState());
+    requestAnimationFrame(() => body.classList.remove("notransition"))
+
+    const toggleTheme = () => {
+        const state = getThemeState();
+        if (state === THEMES.DARK) {
+            localStorage.setItem(LS_THEME_KEY, THEMES.LIGHT);
+            initTheme(THEMES.LIGHT);
+        } else if (state === THEMES.LIGHT) {
+            localStorage.setItem(LS_THEME_KEY, THEMES.DARK);
+            initTheme(THEMES.DARK);
+        }
+    }
+
+    window.addEventListener('DOMContentLoaded', () => {
+        // Theme switch
+        const lamp = document.getElementById("mode");
+
+        lamp.addEventListener("click", () => toggleTheme());
+
+        // Blur the content when the menu is open
```

Comment:
This can be replaced with CSS.
Unfortunately `:has()` is still experimental and disabled by default in `firefox` and unavailable in `firefox android`.
See:  [MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/:has#browser_compatibility) and [caniuse](https://caniuse.com/?search=%3Ahas)

So I'm not sure is this a good idea to use it.



```scss
body:has(#menu-trigger:checked) {
    .wrapper {
      animation: 0.2s ease-in forwards blur;
      -webkit-animation: 0.2s ease-in forwards blur;
    }
}
```